### DISCUSSION TITLE
Are there any examples of building applications?

### DISCUSSION THREAD

**Discussion by bradparks** [webfx-project/webfx#29] [Q&A]
This looks incredible to me, first off 😄 
Are there any demos apps that show how to do a typical app that isn't a game or interactive infographic, something that really tries to show how an app can be deployed with WebFx ?

Examples, I'd love would include a typical "ToDo MVC" app. Thanks!

**Comment by bradparks** [comment]
NOTE: I've seen this older question, on a related topic
external JavaFX libraries. #17

**Comment by salmonb** [comment]
I apologize for the delay; GitHub didn’t notify me about your input, and I discovered it by chance only today!... 🤷

The only real app based on WebFX that I am aware of is [Modality](https://github.com/modalityone/modality) (as I am working on it!). We plan to provide a demo by the end of this year to show the potential of WebFX for enterprise development.

The development of Modality significantly contributes to making WebFX more mature for real app development, including very large applications. Modality currently has over 100 Java modules, all managed by the WebFX CLI, and the WebFX Stack, which includes another 100+ Java modules, will offer a full-stack solution based on Vert.x on the server side. From this perspective, WebFX + Vert.x is similar to React + Node.js, but in Java.

Regarding the usage of existing JavaFX libraries, they will likely not work with WebFX as they were not designed to run in the browser sandbox, unless they are ported to WebFX (which is the purpose of WebFX Libs). However, WebFX Extras, another library with over 50 Java modules that we provide, can be used for WebFX apps. Modality actually uses only these two libraries: WebFX Stack and WebFX Extras.

Most of my time is currently spent on Modality, which is our priority. I will probably need to wait until the first version of Modality is finished before I can create WebFX tutorials on building real apps, likely next year. In the meantime, I can provide support through the [WebFX Discord chat](https://discord.gg/xJcvmGvqh9) for developers who want to try WebFX before these tutorials are available.

**Reply by bradparks** [reply]
No worries, and thanks for the info - much appreciated!

**Comment by ivan-velikanov** [comment]
@salmonb Could you please share a simpler webapp demo application (compared to [Modality](https://github.com/modalityone/modality)) ? Sometime like simple chatgpt kind of chat interface (something more simpler than this - https://github.com/jesuino/LLMFX ) but with a few  views/pages, some basic routing, vertx backed all those aspects, but a bare minimum simple version? It would be really helpful we would be able to pickup much faster, especially if your project size is kept small we could load it in one of the llm/AIs it would be easy fast and instantly productive way to get started. 
Right now I have my full stack in java, right from native (using ffm) to web (using teavm), scripts using single file java or jbang, database using arcadedb, even html with a DSL in java (good for static websites and some dynamic aspects with teavm). 
Now the only thing remaining is being able to make a bit more serious and larger webapp using WebFX / JavaFX. Once this happens I am in full java utopia.
